ipa-core/src/protocol/ipa_prf/aggregation/mod.rs

```diff
@@ -503,37 +501,38 @@ pub mod tests {
                                       (
                                           len in 0..=max_len,
                                           tv_bits in 0..=PROP_MAX_TV_BITS,
-                                          seed in any::<u64>(),
+                                      )
+                                      (
+                                          len in Just(len),
+                                          tv_bits in Just(tv_bits),
+                                          inputs in prop::collection::vec(prop::array::uniform(0u32..1 << tv_bits), len),
                                       )
         -> AggregatePropTestInputs {
-            let mut rng = StdRng::seed_from_u64(seed);
             let mut expected = vec![0; PROP_BUCKETS];
-            let inputs = repeat_with(|| {
-                let row: [u32; PROP_BUCKETS] = array::from_fn(|_| rng.gen_range(0..1 << tv_bits));
+            for row in &inputs {
                 for (exp, val) in expected.iter_mut().zip(row) {
                     *exp = min(*exp + val, (1 << PropHistogramValue::BITS) - 1);
                 }
-                row
-            })
-            .take(len)
-            .collect();
+            }
```

Comment:
These changes are not essential to this PR, but it seemed clearer not to have two different seeds, and using proptest for input generation enables using its other capabilities like shrinking.